### PR TITLE
feat(ui-react-storage): Added upload actions and option to upload files on click

### PR DESCRIPTION
**Pull request by ioanabrooks**
#### Description of changes

* Added Storage Manager footer subcomponent with upload actions and option to turn off default behavior of auto-uploading files. 
* Added unit and integration tests
* Updated docs

| With one file selected  | After click on "Upload 1 file" |
| ------------- | ------------- |
|  Screen Shot 2023-08-22 at 10 54 30 AM | Screen Shot 2023-08-22 at 10 55 20 AM|  

**Docs**

Screen Shot 2023-08-22 at 1 02 21 PM

#### Issue #, if available

Fixes: https://github.com/aws-amplify/amplify-ui/issues/3796

#### Checklist

- [ ] Have read the [Pull Request Guidelines](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md)
- [ ] PR description included
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] `yarn test` passes and tests are updated/added
- [ ] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/react-storage/src/components/StorageManager/StorageManager.tsx [hunk 1]
```diff
@@ -25,6 +26,7 @@ function StorageManagerBase(
   {
     acceptedFileTypes = [],
     accessLevel,
+    autoUpload = true,
```
Comment (ioanabrooks) [inline packages/react-storage/src/components/StorageManager/StorageManager.tsx:29]: Optional prop with default set to true, which is the current behavior (no braking change)

File: packages/e2e/features/ui/components/storage/storage-manager/upload-file-to-S3-public-upload-actions.feature [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    Then I see "Browse files"
+    Then I select a file with file name "test.jpg"
+    Then I see "test.jpg"
+    Then I see "Upload 1 file"
+    Then I see "Clear all"
+    Given I intercept '{ "method": "POST", "url": "**/test.jpg?uploads"  }' with fixture "Storage.public-uploads.xml"
+    Given I intercept '{ "method": "PUT", "url": "**/test.jpg?partNumber=1**"  }' with fixture "Storage.public-upload-part.xml" and add header "Etag" with value "&quot;abc123&quot;"
+    Given I intercept '{ "method": "POST", "url": "**/test.jpg?uploadId**"  }' with fixture "Storage.public-upload-complete-multipart.xml"
+    Given I intercept '{ "method": "GET", "url": "**/?list-type=2**"  }' with fixture "Storage.public-uploads-list.xml"
+    Then I click the "Upload 1 file" button
```
Comment (ioanabrooks) [inline packages/e2e/features/ui/components/storage/storage-manager/upload-file-to-S3-public-upload-actions.feature:17]:
```suggestion
    Then I see "1 file selected"
    Then I click the "Upload 1 file" button
```

File: packages/ui/src/theme/__tests__/overrides.test.ts [hunk 3]
```diff
@@ -1067,12 +1067,7 @@ describe('@aws-amplify/ui', () => {
         --amplify-components-storagemanager-previewer-body-padding-block: var(--amplify-space-medium);
         --amplify-components-storagemanager-previewer-body-padding-inline: var(--amplify-space-medium);
         --amplify-components-storagemanager-previewer-body-gap: var(--amplify-space-small);
-        --amplify-components-storagemanager-previewer-footer-border-color: var(--amplify-colors-border-secondary);
-        --amplify-components-storagemanager-previewer-footer-border-style: solid;
-        --amplify-components-storagemanager-previewer-footer-border-width: var(--amplify-border-widths-small);
-        --amplify-components-storagemanager-previewer-footer-padding-block: var(--amplify-space-medium);
-        --amplify-components-storagemanager-previewer-footer-padding-inline: var(--amplify-space-medium);
-        --amplify-components-storagemanager-previewer-footer-justify-content: space-between;
```
Comment (ioanabrooks) [inline packages/ui/src/theme/__tests__/overrides.test.ts:1075]: These were not used in the first place, storage manager did not have a footer. With the current UX borders don't look good so didn't need them.

**Review by dbanksdesign — APPROVED** [review]
LGTM!

**Review by hbuchel — APPROVED** [review]
Checked it out locally and this LGTM!

**Pull request merged**